Drift/Private/DriftBase.cpp

```diff
@@ -4639,6 +4644,59 @@ void FDriftBase::GetUserIdentitiesByName(const FString& name, const FDriftGetUse
     GetUserIdentitiesByNames(namesArray, delegate);
 }
 
+void FDriftBase::GetFriendRichPresence(int32 FriendId, const FDriftGetFriendRichPresenceDelegate& Delegate)
+{
+    if (state_ != DriftSessionState::Connected)
+    {
+        DRIFT_LOG(Base, Warning, TEXT("Attempting to get rich presence without being connected"));
+        (void) Delegate.ExecuteIfBound(false, {});
+        return;
+    }
+
+    FString Url = driftEndpoints.template_richpresence.Replace(TEXT("{player_id}"), *FString::FromInt(FriendId));
+
+    const auto Request = GetGameRequestManager()->Get(Url);
+    Request->OnResponse.BindLambda([this, Delegate, FriendId](ResponseContext& Context, JsonDocument& Doc)
+    {
+        FRichPresenceResult Result;
+        if (!JsonArchive::LoadObject(Doc, Result))
+        {
+            Context.error = TEXT("Failed to parse user identities response");
+            return;
+        }
+
+        RichPresenceCache.Add(FriendId, Result);
+        (void) Delegate.ExecuteIfBound(true, Result);
+    });
+
+    Request->OnError.BindLambda([this, Delegate, Url](ResponseContext& Context)
+    {
+        FString Error;
+        Context.errorHandled = GetResponseError(Context, Error);
+        DRIFT_LOG(Base, Error, TEXT("Failed to get richpresence from url: %s. Error: '%s'"), *Url, *Error);
```

Comment:
I'd suggest using `IErrorReporter::Get()->AddError(...)` here to make sure this issue would pop up on Sentry.

See this slack message for info
https://directive.slack.com/archives/C02BZ0SQD/p1718808451847629